test/export/test_export.py

```diff
@@ -8287,6 +8287,76 @@ def forward(self, arg0_1: "f32[2, 4]", arg1_1: "f32[4]"):
 """,
         )
 
+    def test_export_for_training_with_state_dict_hooks(self):
+        def _state_dict_pre_hook(mod, prefix, keep_vars):
+            mod._buffers["test"] = torch.Tensor([1])
+
+        def _state_dict_hook(mod, state_dict, prefix, *args, **kwargs):
+            keys = list(state_dict.keys())
+            for key in keys:
+                local_key = key[len(prefix) :]
+                if local_key.startswith("linear"):
+                    new_key = prefix + local_key.replace("linear", "lin")
+                    state_dict[new_key] = state_dict[key]
+                    if new_key != key:
+                        del state_dict[key]
+
+        class CustomModule(torch.nn.Module):
```

Comment:
maybe this test case could match what we were seeing before with more modules? Like if a user is trying to remove a layer out of the state dict

Reply:
> maybe this test case could match what we were seeing before with more modules? Like if a user is trying to remove a layer out of the state dict

fixed now.